src/test/java/plugins/WorkflowPluginTest.java

```diff
@@ -158,4 +168,31 @@
         build.shouldContainsConsoleOutput("divided into 3 sets");
     }
 
+    @WithDocker
+    @WithPlugins({"workflow-aggregator@1.14", "docker-workflow@1.14-SNAPSHOT", "git", "ssh-agent@1.10-SNAPSHOT"})
+    @WithCredentials(credentialType=WithCredentials.SSH_USERNAME_PRIVATE_KEY, values={"git", "/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/unsafe"}, id="gitcreds")
+    @Issue("JENKINS-27152")
+    @Test public void sshGitInsideDocker() throws Exception {
+        // Pending https://github.com/jenkinsci/docker-workflow-plugin/pull/31 (and additional APIs in DockerFixture etc.)
+        // it is not possible to run this on a bind-mounted Docker slave, so we cannot verify that paths are local to the slave.
+        GitContainer container = gitServer.get();
+        String host = container.host();
+        int port = container.port();
+        GitRepo repo = new GitRepo();
+        repo.commit("Initial commit");
+        repo.transferToDockerContainer(host, port);
+        WorkflowJob job = jenkins.jobs.create(WorkflowJob.class);
+        job.script.set(
+            "node {ws('" + tmp.getRoot() + "') {\n" + // TODO UNIX_PATH_MAX workaround
+            "  docker.image('cloudbees/java-build-tools').inside {\n" +
+            "    git url: '" + container.getRepoUrlInsideDocker() + "', credentialsId: 'gitcreds'\n" +
+            "    sh 'mkdir ~/.ssh && echo StrictHostKeyChecking no > ~/.ssh/config'\n" +
```

Comment:
There are no private files in this directory, only configuration, and anyway it is just a test.